### PR TITLE
Oauth

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/oauth-api/api-changes.json [hunk 1]
```diff
@@ -1,7 +1,40 @@
 {
+  "4.3.0": {
```
Comment (afelisatti) [inline modules/oauth-api/api-changes.json]: You'll likely need to change this later once it's cherry-picked.

File: tests/performance/src/main/java/org/mule/MultiMapBenchmark.java [hunk 2]
```diff
@@ -30,6 +31,7 @@ public void setup() throws Exception {
   }
 
   @Benchmark
+  @OutputTimeUnit(NANOSECONDS)
```
Comment (afelisatti) [inline tests/performance/src/main/java/org/mule/MultiMapBenchmark.java]: Remove

File: modules/oauth-api/api-changes.json (hunk 1 above)
Comment (marianogonzalez) [reply modules/oauth-api/api-changes.json]: indeed

File: tests/performance/src/main/java/org/mule/MultiMapBenchmark.java (hunk 2 above)
Comment (marianogonzalez) [reply tests/performance/src/main/java/org/mule/MultiMapBenchmark.java]: i suck

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/UpdatingAuthorizationCodeState.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+/**
+ * An implementation of {@link AuthorizationCodeListener} which registers an {@link AuthorizationCodeListener}
+ * in order to get updated state when a refresh token operation is completed or the resource is simply re-authorized.
+ * 
+ */
+public class UpdatingAuthorizationCodeState implements AuthorizationCodeState {
+
+  private AuthorizationCodeState delegate;
+  private final Consumer<ResourceOwnerOAuthContext> onUpdate;
```
Comment (sonar-mulesoft) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/UpdatingAuthorizationCodeState.java]: Code Smell Code Smell: Remove this unused "onUpdate" private field. ([squid:S1068](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1068))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7744&issues=AWmiFXZHotIXnmsUruQ4&open=AWmiFXZHotIXnmsUruQ4)

[ISSUE_KEY]: AWmiFXZHotIXnmsUruQ4

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/OAuthConnectionProviderWrapper.java [hunk 4]
```diff
@@ -64,12 +66,16 @@ public C connect() throws ConnectionException {
...
   }
 
-  public void updateAuthState() {
-    ResourceOwnerOAuthContext context = getContext();
-
+  private void updateAuthState() {
     final ConnectionProvider<C> delegate = getDelegate();
-    authCodeStateSetter.set(delegate, toAuthorizationCodeState(oauthConfig, context));
+    ResourceOwnerOAuthContext context = getContext();
+    authCodeStateSetter.set(delegate, new UpdatingAuthorizationCodeState(oauthConfig, dancer, context, updatedContext -> {
```
Comment (sonar-mulesoft) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/OAuthConnectionProviderWrapper.java]: Code Smell Code Smell: Remove useless curly braces around statement ([squid:S1602](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1602))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7744&issues=AWmiFXYFotIXnmsUruQ3&open=AWmiFXYFotIXnmsUruQ3)

[ISSUE_KEY]: AWmiFXYFotIXnmsUruQ3

**Pull request merged**